src/test/java/uk/gov/companieshouse/api/accounts/service/impl/IntangibleAssetsServiceTest.java

```diff
@@ -0,0 +1,331 @@
+
+    @Mock
+    private IntangibleAssetsTransformer transformer;
+
+    @Mock
+    private IntangibleAssetsRepository repository;
+
+    @Mock
+    private SmallFullService smallFullService;
+
+    @Mock
+    private KeyIdGenerator keyIdGenerator;
+
+    @Mock
+    private IntangibleAssets intangibleAssets;
+
+    @Mock
+    private Transaction transaction;
+
+    @Mock
+    private TransactionLinks transactionLinks;
+
+    @Mock
+    private HttpServletRequest request;
+
+    @Mock
+    private IntangibleAssetsEntity intangibleAssetsEntity;
+
+    @Mock
+    private Map<String, String> links;
+
+    @InjectMocks
+    private IntangibleAssetsService intangibleAssetsService;
+
+    private static final String COMPANY_ACCOUNTS_ID = "companyAccountsId";
+    private static final String GENERATED_ID = "generatedId";
+    private static final String SELF_LINK = "selfLink";
+
+    @Test
+    @DisplayName("Tests the successful creation of a Intangible Assets resource")
+    void createIntangibleAssetsSuccess() throws DataException {
+
+
+        when(transformer.transform(intangibleAssets)).thenReturn(intangibleAssetsEntity);
+        when(keyIdGenerator.generate(COMPANY_ACCOUNTS_ID + "-" + ResourceName.INTANGIBLE_ASSETS.getName()))
+                .thenReturn(GENERATED_ID);
+
+        when(intangibleAssets.getLinks()).thenReturn(links);
+        when(links.get(BasicLinkType.SELF.getLink())).thenReturn(SELF_LINK);
+
+        when(transaction.getLinks()).thenReturn(transactionLinks);
+        when(transactionLinks.getSelf()).thenReturn(SELF_LINK);
+
+        ResponseObject<IntangibleAssets> response =
+                intangibleAssetsService.create(intangibleAssets, transaction, COMPANY_ACCOUNTS_ID, request);
+
+        assertMetaDataSetOnRestObject();
+        assertIdGeneratedForDatabaseEntity();
+        assertRepositoryInsertCalled();
+        assertWhetherSmallFullServiceCalledToAddLink(true);
+        assertEquals(ResponseStatus.CREATED, response.getStatus());
+        assertEquals(intangibleAssets, response.getData());
+    }
+
+    @Test
+    @DisplayName("Tests the creation of a Intangible Assets resource where the repository throws a duplicate key exception")
+    void createIntangibleAssetsDuplicateKeyException() throws DataException {
+
+        when(transformer.transform(intangibleAssets)).thenReturn(intangibleAssetsEntity);
+        when(keyIdGenerator.generate(COMPANY_ACCOUNTS_ID + "-" + ResourceName.INTANGIBLE_ASSETS.getName()))
+                .thenReturn(GENERATED_ID);
+        when(repository.insert(intangibleAssetsEntity)).thenThrow(DuplicateKeyException.class);
+
+        when(transaction.getLinks()).thenReturn(transactionLinks);
+        when(transactionLinks.getSelf()).thenReturn(SELF_LINK);
+
+        ResponseObject<IntangibleAssets> response =
+                intangibleAssetsService.create(intangibleAssets, transaction, COMPANY_ACCOUNTS_ID, request);
+
+        assertMetaDataSetOnRestObject();
```

Comment:
Nitpicky: Since the entity is never stored in the DB, there's not much point in asserting that data has been set on the object; unit tests should, in my opinion, assert behaviour relevant to the context. Happy to be challenged on this point if you feel otherwise 👍

Reply:
You're correct, assertion removed.